include/dxc/DXIL/DxilResourceProperties.h

```diff
@@ -0,0 +1,82 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// DxilHandleAnnotation.h                                                    //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Representation properties for DXIL handle.                                //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "DxilConstants.h"
+
+namespace llvm {
+class Constant;
+class Type;
+}
+
+namespace hlsl {
+
+struct DxilResourceProperties {
+  DXIL::ResourceClass Class;
+  DXIL::ResourceKind  Kind;
+  static constexpr unsigned kSampleCountUndefined = 0x7;
+
+  union {
+    struct {
+      DXIL::ComponentType CompType : 5; // TypedBuffer/Image.
+      unsigned SingleComponent : 1; // Return type is single component.
+      // 2^SampleCountPow2 for Sample count of Texture2DMS.
+      unsigned SampleCountPow2 : 3;
+      unsigned Reserved : 23;
+    } Typed;
+    unsigned ElementStride; // in bytes for StructurizedBuffer.
+    DXIL::SamplerFeedbackType SamplerFeedbackType; // FeedbackTexture2D.
+    unsigned SizeInBytes; // Cbuffer instance size in bytes.
+    uint32_t RawDword0;
+  };
+
+  union {
+    struct {
+      unsigned bROV : 1;              // UAV
+      unsigned bGloballyCoherent : 1; // UAV
+      unsigned Reserved : 30;
+    } UAV;
+    uint32_t RawDword1;
+  };
+
+  bool operator==(const DxilResourceProperties &);
+  bool operator!=(const DxilResourceProperties &);
+};
+
```

Comment:
Let's put a static_assert here to catch struct size change to remind us to update this in shader model translation etc...